webapp/controller/PageOverview.controller.js

```diff
@@ -0,0 +1,199 @@
+sap.ui.define([
+	"sap/ui/core/mvc/Controller",
+	'sap/ui/model/json/JSONModel',
+	"sap/ui/model/Filter",
+	"sap/ui/model/FilterOperator",
+	'sap/m/Token',
+], function (Controller, JSONModel, Filter, FilterOperator, Token) {
+	"use strict";
+
+	return Controller.extend("products.app.controller.PageOverview", {
+		
+		onInit: function() {
+			const oModel = new JSONModel();
+			oModel.loadData("../model/products.json");
+		
+			this.getView().setModel(oModel);
+		},
+
+		onAfterRendering: function() {
+			const oMultiInput = this.byId("multiInput");
+			oMultiInput.attachTokenUpdate((event) => {
+				if (event.getParameter("type") === "removed") {
+					const oModel = this.getView().getModel();
+					const aSuppliers = oModel.getProperty("/Suppliers");
+					
+					const aRemovedTokens = event.getParameter("removedTokens");
+				
+					const aRemainingTokens = oMultiInput.getTokens().filter(function (token) {
+					  return !aRemovedTokens.includes(token);
+					});
+					
+					const aSelectedTokens = aRemainingTokens.map(el => el.getText());
+					const aSelectedId = aSuppliers
+											.filter((el) => aSelectedTokens.includes(el.SuppliersName))
+											.map((el) => el.SupplierId);
+					
+					this.onFilter(aSelectedId);
+				}
+			})
+		},
+
+		getSuppliersName: function(data) {
+			const oModel = this.getView().getModel();
+			const aSuppliers = oModel.getProperty("/Suppliers");			
+			
+			return aSuppliers
+					.filter((el) => data.includes(el.SupplierId))
+					.map(el => el.SuppliersName)
+					.join(", ");
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
```

Comment:
You should never get data from controls directly. You should use model instead and get data from it.

something like that in xml:
```xml
<m:SearchInput value="{ui>/querySearch}"/>
```

something like that in controller:
```js
const sQuerySearch = this.getView().getModel("ui").getProperty("/querySearch")
```

Don't forget to initialise ui model